lib/rubocop/minitest/support.rb

```diff
@@ -1,10 +1,259 @@
+    # companion method `assert_no_offenses`. This method is a much
+    # simpler assertion since it just inspects the source and checks
+    # that there were no offenses. The `assert_offense` method has
+    # to do more work by parsing out lines that contain carets.
+    #
+    # If the code produces an offense that could not be autocorrected, you can
+    # use `assert_no_corrections` after `assert_offense`.
+    #
+    # @example `assert_offense` and `assert_no_corrections`
+    #
+    #   assert_offense(<<~RUBY)
+    #     class FooTest < Minitest::Test
+    #       def test_do_something
+    #         assert_equal(nil, somestuff)
+    #         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Prefer using `assert_nil(somestuff)`.
+    #       end
+    #     end
+    #   RUBY
+    #
+    #   assert_no_corrections
+    #
+    class Test < ::Minitest::Test
+      class << self
+        # Helper to define a test method using a String. Under the hood, it replaces
+        # spaces with underscores and defines the test method.
+        #
+        #   test "verify something" do
+        #     ...
+        #   end
+        def test(name, &block)
+          test_name = :"test_#{name.gsub(/\s+/, '_')}"
+          defined = method_defined?(test_name)
+          raise "#{test_name} is already defined in #{self}" if defined
+
+          if block
+            define_method(test_name, &block)
+          else
+            define_method(test_name) do
+              raise("No implementation provided for #{name}")
+            end
+          end
+        end
+      end
+
+      private
+
+      def setup
+        cop_name = self.class.to_s.delete_suffix('Test')
+
+        cop_class = begin
+          Object.const_get("RuboCop::Cop::Minitest::#{cop_name}")
+        rescue NameError # rubocop:disable Lint/SuppressedException
+        end
+
+        cop_class ||= begin
+          Object.const_get(cop_name)
+        rescue NameError # rubocop:disable Lint/SuppressedException
+        end
+
+        @cop = cop_class.new(configuration) if cop_class&.ancestors&.include?(RuboCop::Cop::Base)
+      end
+
+      def format_offense(source, **replacements)
+        replacements.each do |keyword, value|
+          value = value.to_s
+          source = source.gsub("%{#{keyword}}", value)
+                         .gsub("^{#{keyword}}", '^' * value.size)
+                         .gsub("_{#{keyword}}", ' ' * value.size)
+        end
+        source
+      end
+
+      def assert_no_offenses(source, file = nil)
+        setup_assertion
+
+        offenses = inspect_source(source, @cop, file)
+
+        expected_annotations = RuboCop::RSpec::ExpectOffense::AnnotatedSource.parse(source)
+        actual_annotations = expected_annotations.with_offense_annotations(offenses)
+
+        assert_equal(source, actual_annotations.to_s)
+      end
+
+      def assert_offense(source, file = nil, **replacements)
+        setup_assertion
+
+        @cop.instance_variable_get(:@options)[:autocorrect] = true
```

Comment:
If the testfile isn't named properly this results in a `NoMethodException`

Reply:
Fixed, I made `setup_assertion` raise if there's no `@cop` defined.